scan-command/src/main/java/io/ballerina/scan/internal/ReporterImpl.java

```diff
@@ -78,19 +78,19 @@ private Issue createIssue(Document reportedDocument, Location location, Rule rul
         String[] parts = fullyQualifiedRuleId.split(":");
 
         Source source;
-        if (parts[0].equals(BALLERINA_RULE_PREFIX + rule.numericId())) {
+        if (parts[0].equals(BALLERINA_RULE_PREFIX) && parts[1].equals(String.valueOf(rule.numericId()))) {
```

Comment:
We generally do checks like `BALLERINA_RULE_PREFIX.equals(parts[0])` to avoid potential NPEs if they are a concern.